public/app/features/canvas/runtime/scene.tsx

```diff
@@ -426,9 +436,25 @@ export class Scene {
       .on('dragStart', (event) => {
         this.ignoreDataUpdate = true;
         this.setNonTargetPointerEvents(event.target, true);
+
+        if (this.moveable && this.moveable.elementGuidelines) {
+          const targetIndex = this.moveable.elementGuidelines.indexOf(event.target);
```

Comment:
Nit: may not be worth it, but there's some duped code a few time here that we could maybe abstract out? Either way, this is awesome!

Reply:
ah yes that is a good point, a lot of repeated code here, thanks for pointing this out!

Reply:
on closer review breaking out this logic I think is not really in scope of these changes (honestly the `scene.tsx` file has been needing a cleaning up for a while now (ideally breaking out moveable / selecto config to their own file where we can better group helper functions specific to setting up config (and also cleaning up `scene.tsx` a ton)

For now I've included some comments to help with future readability / maintainability 🤞